cmd/tk/workflow.go

```diff
@@ -101,15 +118,26 @@ func diffCmd() *cobra.Command {
 			kube.Spec.DiffStrategy = *diffStrategy
 		}
 
-		desired, err := kube.Reconcile(raw, stringsToRegexps(vars.targets)...)
+		desired, err := kube.Reconcile(raw, environment, stringsToRegexps(vars.targets)...)
 		if err != nil {
 			log.Fatalln("Reconciling:", err)
 		}
 
-		if diff(desired, interactive, kubernetes.DiffOpts{Summarize: *summarize}) {
-			os.Exit(16)
+		if !diff(desired, interactive, kubernetes.DiffOpts{Summarize: *summarize}) {
+			log.Println("No differences.")
+		}
+
+		if *willDelete {
+			deletions, err := kube.ReconcileDeletions(desired, kubernetes.ReconcileDeletionsOpts{Environment: environment})
+			if err != nil {
+				log.Fatalln("Reconciling deletions:", err)
+			}
+			for _, deletion := range deletions {
+				fmt.Println("Requires deletion: ", deletion)
+			}
 		}
-		log.Println("No differences.")
+
+		os.Exit(16)
```

Comment:
This will result in the process always exiting with code 16, even if there are no diffs.